File: app/src/main/java/com/nicholasdoglio/eyebleach/model/reddit/Image.java
package com.nicholasdoglio.eyebleach.model.reddit;


import com.squareup.moshi.Json;

import java.util.List;

public class Image {


    @Json(name = "source")
    private Source source;
    @Json(name = "resolutions")
    private List<Resolution> resolutions = null;
    @Json(name = "variants")
    private Variants variants;
    @Json(name = "id")
    private String id;

    public Source getSource() {
        return source;
    }

    public void setSource(Source source) {
        this.source = source;
    }

    public List<Resolution> getResolutions() {
        return resolutions;
    }

    public void setResolutions(List<Resolution> resolutions) {
        this.resolutions = resolutions;
    }

    public Variants getVariants() {
        return variants;
    }

    public void setVariants(Variants variants) {
        this.variants = variants;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

}

File: remote/src/main/java/com/nicholasdoglio/remote/entities/Image.kt
package com.nicholasdoglio.remote.entities

import com.squareup.moshi.Json

data class Image(
  @Json(name = "source") val source: Source,
  @Json(name = "resolutions") val resolutions: List<Resolution>,
  @Json(name = "variants") val variants: Variants,
  @Json(name = "id") val id: String // PY4ZzYSGatREt9dMgLav0_eqy4cn1z5bAx4Z3RSt7is
)